oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/DocumentStoreIndexerBase.java

```diff
@@ -78,15 +80,62 @@ protected void setProviders() throws IOException {
         this.indexerProviders = createProviders();
     }
 
+    private static class MongoNodeStateEntryTraverserFactory implements NodeStateEntryTraverserFactory {
+
+        private static final AtomicInteger traverserInstanceCounter = new AtomicInteger(0);
+        final RevisionVector rootRevision;
+        final DocumentNodeStore documentNodeStore;
+        final MongoDocumentStore documentStore;
+        final Logger traversalLogger;
+        final CompositeIndexer indexer;
+        final Closer closer;
```

Comment:
I think all of those fields can also be private, right (didn't verify)?

Reply:
yes. done.